Notation.Plugin.AzureKeyVault/KeyVault/KeyVaultClient.cs

```diff
@@ -115,30 +112,36 @@ private static KeyVaultMetadata ParseId(string id)
         {
             if (string.IsNullOrEmpty(id))
             {
-                throw new ArgumentNullException(nameof(id), "Id must not be null or empty");
+                throw new ValidationException("Input passed to \"--id\" must not be empty");
             }
 
             var uri = new Uri(id);
             // Validate uri
-            if (uri.Segments.Length != 4)
+            if (uri.Segments.Length < 3 || uri.Segments.Length > 4)
             {
-                throw new ValidationException(INVALID_INPUT_ERROR_MSG);
+                throw new ValidationException("Invalid input passed to \"--id\". Please follow this format to input the ID \"https://{vault-name}.vault.azure.net/certificates/{certificate-name}\"");
```

Comment:
I assume this error message prompts users use the latest key. Will this error message narrows to a single scenario, i.e. always to use the latest key? What if users want to specify a version of key?

Reply:
When the ID cannot be parsed at all, we will return this error message. This means we cannot distinguish whether the user wants to use a versioned or versionless ID.

Updated the error message to include versioned ID.

Reply:
>          throw new ValidationException("Invalid input passed to \"--id\". Please follow this format to input the ID \"https://{vault-name}.vault.azure.net/certificates/{certificate-name}\" or \"https://{vault-name}.vault.azure.net/certificates/{certificate-name}/{certificate-version}\"");

Reply:
LGTM. Thank you @JeyJeyGao